### ISSUE TITLE
[Question] - GHE support

### ISSUE THREAD

**Issue by bondezbond** [WyriHaximus/github-action-wait-for-status#96]
I couldn't find anything in src but I wanted to check to see if this works for GHE?

**Comment by WyriHaximus** [comment]
Hmmm good point, I'll look into picking up the API base URL from environment variables. Or by making it configurable.

**Comment by sdevineni** [comment]
@WyriHaximus can you please help addressing this issue for GHE. this is a very good action to have it working on GHE

**Comment by WyriHaximus** [comment]
Working on it in #124

**Comment by WyriHaximus** [comment]
Just released [`v1.7.0`](https://github.com/WyriHaximus/github-action-wait-for-status/releases/tag/v1.7.0). Please let me know if this makes it work on GHE for you.

**Comment by sdevineni** [comment]
Still ended up with a similar issue as before

Uncaught Throwable ApiClients\Tools\Psr7\HttpStatusExceptions\UnauthorizedException: "Unauthorized" at /workdir/vendor/api-clients/psr7-http-status-exceptions/src/UnauthorizedException.php line 12

[2022-09-23 14:18:03.300124] wait.DEBUG: Error reason: {"message":"Bad credentials","documentation_url":"https://docs.github.com/rest"}

**Comment by WyriHaximus** [comment]
> Still ended up with a similar issue as before

Will double check some things later, but what was the error you got before?

**Comment by WyriHaximus** [comment]
Right so #124 made it use `GITHUB_API_URL` to get the API URL. Confirmed it uses that during development, and I just did that again to be absolutely sure it does use that. It should solve your issues as I'm unaware of any other differences between Public GitHub.com and GHE.

**Comment by WyriHaximus** [comment]
@sdevineni Have you tried using `@v1.7.0` to force it to that version?

**Comment by sdevineni** [comment]
yes we tried pulling in the code and pointing to it.  will try using it directly from this repo

**Comment by sdevineni** [comment]
[2022-09-23 16:26:23.997513] wait.ERROR: Uncaught Throwable RuntimeException: "Connection to tls://github.infra.xxx.com:443 failed during DNS lookup. Last error for IPv6: DNS query for github.infra.xxx.com (AAAA) returned an error response (Non-Existent Domain / NXDOMAIN). Previous error for IPv4: DNS query for github.infra.xxx.com (A) returned an error response (Non-Existent Domain / NXDOMAIN)" at /workdir/vendor/react/socket/src/SecureConnector.php line 76
[2022-09-23 16:26:23.997599] wait.INFO: Final status: error

**Comment by WyriHaximus** [comment]
Ow hmmm I might know

**Comment by WyriHaximus** [comment]
Yup found it, will put together a fix tonight

**Comment by sdevineni** [comment]
Thanks a lot @WyriHaximus

**Comment by WyriHaximus** [comment]
Thank you for the errors @sdevineni :+1:

**Comment by WyriHaximus** [comment]
> [2022-09-23 16:26:23.997513] wait.ERROR: Uncaught Throwable RuntimeException: "Connection to tls://github.infra.xxx.com:443 failed during DNS lookup. Last error for IPv6: DNS query for github.infra.xxx.com (AAAA) returned an error response (Non-Existent Domain / NXDOMAIN). Previous error for IPv4: DNS query for github.infra.xxx.com (A) returned an error response (Non-Existent Domain / NXDOMAIN)" at /workdir/vendor/react/socket/src/SecureConnector.php line 76 [2022-09-23 16:26:23.997599] wait.INFO: Final status: error

@sdevineni Just tagged [`v1.7.1`](https://github.com/WyriHaximus/github-action-wait-for-status/releases/tag/v1.7.1) that should fix that error

**Comment by sdevineni** [comment]
Will try it out. Please allow me some time.

**Comment by WyriHaximus** [comment]
Keep me posted, it's 2158 here, so the next time I can have a look is Saturday morning

**Comment by swathida-byte** [comment]
[2022-09-25 15:33:22.431628] wait.ERROR: Uncaught Throwable ApiClients\Tools\Psr7\HttpStatusExceptions\NotAcceptableException: "Not Acceptable" at /workdir/vendor/api-clients/psr7-http-status-exceptions/src/NotAcceptableException.php line 12

**Comment by sdevineni** [comment]
@WyriHaximus  we found the above issue, @swathida-byte is from my team.

**Comment by WyriHaximus** [comment]
@swathida-byte What are the lines above it?

@sdevineni Cool, I'm out of the country for work today. But I'll see to improve debugging at which step this happens. Also, which version of GHE ar you on?

**Comment by sdevineni** [comment]
we are on 3.4.5

**Comment by Zyian** [comment]
Wanted to confirm that the "NotAcceptableException" is present on 3.6.2 as well

**Comment by WyriHaximus** [comment]
> Specifically it is failing on the first call to look up the owner and does not continue to look up the repo

Well if it can't look that up due to an API error it can't be sure to reach the repo either. But that might work differently in GHE 🤔 . I'll reach out to see if I can get access to a GHE instance somewhere and test this out myself directly.
